OptionItem.c

```diff
@@ -0,0 +1,181 @@
+/*
+htop - OptionItem.c
+(C) 2004-2011 Hisham H. Muhammad
+Released under the GNU GPLv2, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "OptionItem.h"
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "CRT.h"
+#include "RichString.h"
+
+
+static void CheckItem_delete(Object* cast) {
+   CheckItem* this = (CheckItem*)cast;
+   assert (this != NULL);
+
+   free(this->text);
+   free(this);
+}
+
+static void CheckItem_display(const Object* cast, RichString* out) {
+   const CheckItem* this = (const CheckItem*)cast;
+   assert (this != NULL);
+
+   RichString_write(out, CRT_colors[CHECK_BOX], "[");
+   if (CheckItem_get(this)) {
+      RichString_append(out, CRT_colors[CHECK_MARK], "x");
+   } else {
+      RichString_append(out, CRT_colors[CHECK_MARK], " ");
+   }
+   RichString_append(out, CRT_colors[CHECK_BOX], "] ");
+   RichString_append(out, CRT_colors[CHECK_TEXT], this->text);
+}
+
+static void NumberItem_delete(Object* cast) {
+   NumberItem* this = (NumberItem*)cast;
+   assert (this != NULL);
+
+   free(this->text);
+   free(this);
+}
+
+static void NumberItem_display(const Object* cast, RichString* out) {
+   const NumberItem* this = (const NumberItem*)cast;
+   assert (this != NULL);
+
+   char buffer[12];
+   RichString_write(out, CRT_colors[CHECK_BOX], "[");
+   xSnprintf(buffer, sizeof(buffer), "%d", NumberItem_get(this));
```

Comment:
Allow for sclaed display?

E.g. for the refresh delay it stores 1 internally, but shows 0.1 using a provided scale factor?

Reply:
then it gets complicated, e.g. how to display `0.1` as it is no exact floating point number.

Reply:
Two options:
1. Just use a fixed point display (number of digits behind decimal point provided), display with zero-padding at least `fixed-point+1` digits, insert decimal point at correct place.
2. `xSnprintf("%.*f", decimal_places, (double)number/scale);` (cf. https://stackoverflow.com/a/16413665)
3. Allow for custom display handler to be provided (probably overkill)

Option 2 is probably fully sufficient and easiest to implement.

Reply:
Support for scaling in both directions (`*10`, `/10`) and scaling with non-multiples of 10 will also complicate things.
d
After all is this really needed or can we expect user who want to change the scan interval what deci-seconds are?

Reply:
I think it's more intuitive, if the user sees fractional seconds. I don't think deciseconds is common enough that the average user will understand without mentally stumpbling over it first.

Scaling with just `*10` and `/10` (or powers thereof) should suffice (and works as suggested above).